Ecal prefire fix

cverstege opened the pull request.

HerrHorizontal commented on Compile/src/Producers/PrefiringWeightProducer.cc:
```diff
@@ -125,7 +125,14 @@ void PrefiringWeightProducer::Produce(ZJetEvent const& event,
     //Jets
-    const auto& theJets = product.m_correctedZJets["L1L2L3Res"];
+    auto& validJets = product.m_correctedZJets["L1L2L3Res"];
+    auto& invalidJets = product.m_correctedInvalidJets["L1L2L3Res"];
```
Für den Fall, dass die `L1L2L3Res` Pipeline nicht konfiguriert ist, scheitert das natürlich. Vielleicht sollte man das abfangen.

cverstege replied:
Log Warning weiter unten bei 0 Jets

HerrHorizontal commented on Compile/src/Producers/PrefiringWeightProducer.cc:
```diff
@@ -241,20 +248,20 @@ void PrefiringWeightProducer::Produce(ZJetEvent const& event,
     //Move global prefire weights, as well as those for muons, photons, and jets, to the event
     //Up and Down are switched, since we use the inverse and apply it to data.
     product.m_optionalWeights["prefiringWeight"] =  1./nonPrefiringProba[0];
```
Kann man das nominelle Weight nicht zu dem Event-Weight hinzufügen?

cverstege replied:
Natürlich

HerrHorizontal reviewed (changes requested):
Some minor comments inline.

cverstege commented:
Fixed a bug when calculating the weights. Also moved back to optionalWeights, as those weights are only applied at reco level, not on gen level.
Should be ready now. Please re-review @HerrHorizontal @RHofsaess

HerrHorizontal reviewed (approved):
Luckily, you have spotted the bug. Looks good now! :+1:

RHofsaess commented on Compile/interface/ZJetProduct.h:
```diff
@@ -181,10 +186,22 @@ class ZJetProduct : public KappaProduct
+        }
+        // Uncorrected invalid jet
+        else if (corrLevel == "None") {
+            return static_cast<KLV*>(m_invalidJets[index]);
```
is this really correct?
eventually this is implicitely casted to usigned long when returning?
